requirements.txt

```diff
@@ -25,6 +25,11 @@ Pillow==9.4.0 --global-option="build_ext" --global-option="--disable-tiff" --glo
 flatten-json==0.1.13
 PyYAML==6.0
 Markdown==3.4.1
+black==23.3.0
+isort==5.12.0
+flake8==6.0.0
+mypy==1.4.1
+pre-commit==3.3.3
```

Comment:
If only we used poetry..

And do you need to install these though?
pre-commit creates its own venv for all of these tools, as far as I know

Reply:
Indeed about poetry. Perhaps not needed but it simplifies for developers to install and run them locally I think. But I am open to make changes.

Reply:
It just that you would need to think about versions in pre-commit and here, but we could keep that for now.